torch_xla/torch_xla.py

```diff
@@ -50,3 +52,17 @@ def device_count() -> int:
 def sync():
   """Launches all pending graph operations."""
   xm.mark_step()
+
+
+@contextlib.contextmanager
+def step():
```

Comment:
the reason I find `step` can be a bit confusing is that we don't call `mark_step` upon entering the `step`.

```
with xla.step():
  y = x + z

y += 1
```
`step` as a context kind of suggest execution will only cover what happened in side the context manger but that's actually not the case.

Reply:
I agree. This should either print a warning if there are pending operations, or just mark_step twice. What do you think is better?

Reply:
let's try `mark_step` twice and benchmark it with one of the examples on resneto50 with fakedata.